drivers/xen/pvblock.c

```diff
@@ -40,6 +40,20 @@ struct blkfront_info {
 	int flush;
 };
 
+/**
+ * struct blkfront_dev - Struct representing blkfront device
+ * @dom: Domain id
+ * @ring: Front_ring structure
+ * @ring_ref: The grant references, allowing us to reference
+ *	      the ring pages in the front-end's domain and to
```

Comment:
we are fronted: you are saying that we allowing access to ourselves
what is really happening here we grant access to the ring to the other end/domain